app/assets/javascripts/backbone/views/project_step_view.coffee

```diff
@@ -84,13 +86,21 @@ class MS.Views.ProjectStepView extends Backbone.View
     link = e.currentTarget
     action = @$(link).data('action')
 
+    if @timelineTableView
+      stepId = @stepId
+    else
+      stepId = @$(link).data('parent-step-id')
+
     unless @logModalView
-      @logModalView = new MS.Views.LogModalView(el: $("<div>").appendTo(@$el), parentView: this)
+      if @timelineTableView
+        @logModalView = new MS.Views.LogModalView(el: $('.timeline-table .log-modal'), timelineTableView: @timelineTableView)
+      else
+        @logModalView = new MS.Views.LogModalView(el: $("<div>").appendTo(@$el), parentView: this)
```

Comment:
Note that in this class there are multiple places where you are checking an attribute (`@timelineTableView`) and doing different things depending on it. Consider this an anti-pattern. It indicates that a separate class may be needed or some other kind of refactor.